### ISSUE TITLE
tools: add levels list in Welcome Panel

### ISSUE THREAD

**Issue by dbartolini** [crownengine/crown#112]
To quickly open recently modified levels without having to search them manually every time.

**Comment by SeanKuehl** [comment]
I'm interested in working on this issue. Could you show me exactly what the "welcome page" is in both code and visually? Also providing any potentially relevant files may help.

**Comment by dbartolini** [comment]
Welcome Panel:
![image](https://user-images.githubusercontent.com/2070946/235248757-913f12a2-855b-4c35-b440-f937d64a4469.png)

Welcome Panel code:
https://github.com/crownengine/crown/blob/79cf189900f706a9dfd63fdaf9eb958a4d47a6d1/tools/level_editor/panel_projects_list.vala

This is where the panels are instantiated:
https://github.com/crownengine/crown/blob/79cf189900f706a9dfd63fdaf9eb958a4d47a6d1/tools/level_editor/level_editor.vala#L701-L704

## Levels List
I want to split the welcome panel in half vertically: on the left there's the Projects list while on the right there will be the list containing all the levels in the selected project, ordered in a LRU fashion.
Ideally, the Levels List will offer a simple list view and a grid view (Gtk.IconView maybe?) with thumbnails (you can populate it with stub pictures for now). When you double-click a level it opens in the level editor.
![image](https://user-images.githubusercontent.com/2070946/235257002-7b364897-9889-4c7c-8d03-2b4082118b96.png)

**Comment by SeanKuehl** [comment]
Is there a way to get a list of levels in a project? I looked through Project.vala but couldn't see something like it.

**Comment by dbartolini** [comment]
No, there is not. In the Welcome Panel the data-compiler, which normally serves this purpose, has not started yet.
One could traverse the project's folder manually looking for all files ending with .level extension: https://valadoc.org/gio-2.0/GLib.FileEnumerator.html
